### PR TITLE
Automate daily tests

### PR DESCRIPTION
**Pull request by emlys**
Automated tests on dev.dockstore.net are going to fail for now because they depend on `data-cy` attributes which I added to the UI on this branch

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -204,6 +243,18 @@ workflows:
...
               only: /^develop/
+
+  nightly:
+    triggers:
+      - schedule:
+          cron: "0 11 * * *" # This is 4am PDT / 3am PST
+          filters:
+            branches:
+              only:
+                - feature/SEAB-815/daily-tests
```
Comment (coverbeck) [inline .circleci/config.yml:254]: This needs to be develop

File: cypress/integration/manualTests/monitor.ts [hunk 2]
```diff
@@ -50,3 +52,27 @@ describe('Monitor homepage links', () => {
...
+      .click();
+    cy.get('[data-cy=workflowColumn]')
+      .first()
+      .within(() => {
+        cy.get('a').click(); // click on the link to the first workflow
+      });
+  });
+
+  it('get the svg icons', () => {
+    cy.get('[data-cy=dnanexusIcon]').should('exist');
```
Comment (coverbeck) [inline cypress/integration/manualTests/monitor.ts:74]: Have you tested if this catches the case when the icon is not present?

You can make the network request fail locally by blocking it.

I'm concerned this element may be present if the request to get the icon fails; successfully fetching the icon appends SVG DOM elements, which I think you may need to look for.

File: cypress/integration/manualTests/monitor.ts [hunk 3]
```diff
@@ -50,3 +52,27 @@ describe('Monitor homepage links', () => {
     });
   });
 });
+
+describe('Test launch button icons', () => {
+  it('find a WDL workflow', () => {
+    cy.visit('/search');
+    cy.wait(10000); // wait for the tabs to become clickable
```
Comment (garyluu) [inline cypress/integration/manualTests/monitor.ts:59]: should wait for something specific instead.  like the tab has content (which may imply that it is clickable) or something else.

File: cypress/integration/manualTests/monitor.ts [hunk 4]
```diff
@@ -50,3 +52,27 @@ describe('Monitor homepage links', () => {
...
+    goToTab('Workflows');
+
+    // click to sort by descriptor type descending so WDL is at the top
+    cy.get('[data-cy=descriptorType]')
+      .click()
```
Comment (garyluu) [inline cypress/integration/manualTests/monitor.ts:64]: what's the purpose of the double click?

File: package.json [hunk 5]
```diff
@@ -16,7 +16,8 @@
     "compodoc": "npx compodoc -p src/tsconfig.compodoc.json --output docs",
     "test": "npx ng test",
     "lint": "npx ng lint",
-    "e2e": "npx ng e2e"
+    "e2e": "npx ng e2e",
+    "test-dev": "npx cypress run -c baseUrl=https://dev.dockstore.net -s \"cypress/integration/manualTests/monitor.ts\""
```
Comment (coverbeck) [inline package.json:20]: Can you please add test-staging and test-prod as well. Even though they won't be automated at this point, it will be nice to run the tests manually against staging/prod when we deploy a new release.

File: cypress/integration/manualTests/monitor.ts (hunk 4 above)
Comment (emlys) [reply cypress/integration/manualTests/monitor.ts:64]: the first click sorts by descriptor type ascending, the second click reverses the sort so that the first workflow is WDL

File: cypress/integration/manualTests/monitor.ts (hunk 2 above)
Comment (emlys) [reply cypress/integration/manualTests/monitor.ts]: You're right, should've caught that. Changed it to check for a svg element

File: package.json [hunk 6]
```diff
@@ -79,7 +82,7 @@
     "@types/jquery": "^3.3.31",
     "@types/node": "^10.14.19",
     "codelyzer": "^5.0.1",
-    "cypress": "4.0.1",
+    "cypress": "^4.0.1",
```
Comment (coverbeck) [inline package.json:85]: Why did you change this? It didn't end up changing the package-lock.json (did you verify it didn't?), so it probably doesn't matter, but generally best to avoid unnecessary changes if possible.

Comment (emlys) [reply package.json:85]: I didn't change it, I thought it was changed when I rebuilt it or something.  Can change it back if it's not important

Comment (coverbeck) [reply package.json:85]: I feel like you had to have changed it :) but it's fine
